app/controllers/chats_controller.rb

```diff
@@ -0,0 +1,101 @@
+class ChatsController < ApplicationController
+	before_action :authenticate_user!
+	def index
+		all_recipients
+
+		if @notified_users.count > 0
+			@user = @notified_users.first
+		elsif @followed_users.count > 0
+			@user = @followed_users.first
+		else
+			@user = @all_users.first
+		end
+		@msgs = get_messages(@user)
+		get_read(@msgs)
+	end
+
+	def msg
+		all_recipients
+		
+		@user = User.find(params[:id])
+		@msgs = get_messages(@user)
+		get_read(@msgs)
+	end
+
+	def send_msg
+
+		@user = User.find(params[:chat][:receiver_id])
+		chat = Chat.new(chat_params)
+		chat.sender_id = current_user.id
+		chat.message_status = "unread"
+		
+		if params[:chat][:designs_id].present?
+			chat.message_type = "image"
+			chat.designs_id = params[:chat][:designs_id].split
+		else
+			chat.message_type = "text"
+		end
+
+		if (FollowingList.blocked.where(from_id: params[:chat][:receiver_id], to_id: current_user).count == 0) && (params[:chat][:body] != "" || params[:chat][:designs_id] != "")
```

Comment:
Refactor it. A very long condition and making db query from condition. Move it to a separate method first.